packages/client/src/devices/__tests__/SpeakerManager.test.ts

```diff
@@ -0,0 +1,66 @@
+import { afterEach, beforeEach, describe, vi, it, expect } from 'vitest';
+import { mockAudioDevices } from './mocks';
+import { of } from 'rxjs';
+import { SpeakerManager } from '../SpeakerManager';
+import { checkIfAudioOutputChangeSupported } from '../devices';
+
+vi.mock('../devices.ts', () => {
+  console.log('MOCKING devices');
+  return {
+    getAudioOutputDevices: vi.fn(() => of(mockAudioDevices)),
+    checkIfAudioOutputChangeSupported: vi.fn(() => true),
+  };
+});
+
+describe('SpeakerManager.test', () => {
+  let manager: SpeakerManager;
+
+  beforeEach(() => {
+    manager = new SpeakerManager();
+  });
```

Comment:
This is a nitpick: we should probably start switching to test fixtures to allow for tests to run concurrently which in turn should make them run slightly faster and it looks slightly more nicer. :)

Reply:
I don't know what those are, but sounds good 😄 if you have an example code, I'm happy to start using them

Reply:
https://github.com/GetStream/stream-video-js/blob/96dfb0d5194ddc2efdff8e2ea495cb030775a687/sample-apps/react/egress-composite/tests/layouts.spec.ts#L1-L32

It's these - this is Playwright but I believe Vitest has similar looking API. It runs code within your "extend" definition and provides the data for each test instance defined later. :)